### ISSUE TITLE
Property Mapping:  Exception Error

### ISSUE THREAD

**Issue by UweJakobs** [evoWeb/sf_register#69]
I'm getting the following error with this configuration:

**Comment by garbast** [comment]
Hi Uwe, thanks for your details backtrace. I tried to access the given url but only got this error message 'This version of TYPO3 CMS requires PHP 7.2 or above'

**Comment by UweJakobs** [comment]
Hi Sebastian,
you're a fast guy! And sorry - I was developing something for an older T3-Version under PHP 7.0 an hour ago, please try again, I switched back to 7.2 meanwhile.

**Comment by garbast** [comment]
Okay. So first result is, that the property mapping in general is working. I can state that, because the preview of the entered values are rendered. If the mapping is completely broken even the preview is not accessible anymore.
I tried to get the error on my system, but did not succeed. But i must admit, i use composer for installation.

My guess is, that there is something bugged in the autoloading. Please try to clear all caches with the install tool and recheck.

Without direkt access to the machine i cant diagnose any further.

**Comment by UweJakobs** [comment]
Yes, an autoloading-problem was my first guess too. 
But I deinstalled and reinstalled the Installation, cleared all caches, even deleted the typo3temp-Directory, hit the "Rebuild PHP Autoload Information"-Button about 100 Times. No effect. I'd like to give you full admin-Access to the system, you'll get a PN.

**Comment by garbast** [comment]
I think i fixed it. There were some problem with the argument mapping. Please download the current develop branch for testing and reply if it works for you. If it does i will release a new version shortly after.

**Comment by UweJakobs** [comment]
That was the solution - It works this way!
And you, Sir, are my personal hero from now on... may I print your Avatar on my pillow?

**Comment by UweJakobs** [comment]
Yes, looking better - your solution basically works this way (with a Plugin aleady set).
However, an newly inserted Plugin misses the values in Flexform ("Available Items") and therefore, the field "pi_flexforms" cannot be filled.
Maybe, this is related to the new error " (1/2) #1278450972 TYPO3\CMS\Extbase\Reflection\Exception\UnknownClassException" (see: https://wiki.typo3.org/Exception/CMS/1278450972),  or maybe the error is the reason for the missing items. I'll try to find a solution an let you know.
Maybe, you could check the annotation methods at the same time.

**Comment by UweJakobs** [comment]
The error is discribed as:
(1/2) #1278450972 TYPO3\CMS\Extbase\Reflection\Exception\UnknownClassException
Class Evoweb\SfRegister\Services\Captcha\RecaptchaAdapter does not exist. Reflection failed.

And in fact, I cannot find a Class "RecaptchaAdapter", only a "JmRecaptchaAdapter", but it's registrated properly in the "autoload_classmap.php". I don't know where "RecaptchaAdapter" comes from, and a grep over file system and DB-Dump gave me no hint of an occurance in this spelling.

**Comment by garbast** [comment]
RecaptchaAdapter is only referenced by plugin.tx_sfregister.settings.fields.configuration.captcha.type and should work if EXT:recaptcha is installed.

Just remove the captcha from selected fields and the error should be gone.

**Comment by UweJakobs** [comment]
Yes, that's right. But since the values in Flexform ("Available Items") are not set anymore, you have to do in manually in the "pi_flexform" -field.
And I can conform that this is the main problem now: As soon as the flexform-setting "settings.fields.selected" is filled, everythings works like a charme. In newly added Plugins, where you cannot set the field-Items any more (like gender,title,firstName,lastName,username ...) you will run into the error "(1/2) #1278450972 TYPO3\CMS\Extbase\Reflection\Exception\UnknownClassException
Class Evoweb\SfRegister\Services\Captcha\RecaptchaAdapter does not exist. Reflection failed."  
In this combination, it seems to be a false positive, of course. 
However, if we manage to bring the "itemsProcFunc" back to live, I think, the issue is finally solved (for me).

**Comment by garbast** [comment]
Item proc func is fixed after fixing deprecated function fall and filles fields again.

**Comment by UweJakobs** [comment]
You, Sir, are my personal hero!
Thanks a lot for your fast reply and excellent work!
May I buy you a beer or ten?

**Comment by garbast** [comment]
Instead of buying some beers you could have a look at https://www.amazon.de/registry/wishlist/28QFD89V6A4RY ;)

**Comment by UweJakobs** [comment]
...Gedankenübertragung ;)
Läuft!
